lib/editing/editor.dart

```diff
@@ -56,6 +56,8 @@ abstract class Document {
 
   void select(Position start, [Position end]);
 
+  String get getSelection;
```

Comment:
In idiomatic Dart, you'd either want:

    String get selection;

or

    String getSelection();

And the general contract for getters is that they can run in O(1) time, approximately. This can be amortized; it can to expensive once as long as the average is constant time.

In this case I'd go with a getter, called either `selection` or `currentSelection`.